protocols/relay/src/protocol/outbound_stop.rs

```diff
@@ -161,3 +73,152 @@ pub enum FatalUpgradeError {
     #[error("Unexpected message status '{0:?}'")]
     UnexpectedStatus(proto::Status),
 }
+
+pub(crate) async fn handle_stop_message_response(
+    io: Stream,
+    stop_command: StopCommand,
+    tx: oneshot::Sender<()>,
+) -> handler::RelayConnectionHandlerEvent {
+    let msg = proto::StopMessage {
+        type_pb: proto::StopMessageType::CONNECT,
+        peer: Some(proto::Peer {
+            id: stop_command.src_peer_id.to_bytes(),
+            addrs: vec![],
+        }),
+        limit: Some(proto::Limit {
+            duration: Some(
+                stop_command
+                    .max_circuit_duration
+                    .as_secs()
+                    .try_into()
+                    .expect("`max_circuit_duration` not to exceed `u32::MAX`."),
+            ),
+            data: Some(stop_command.max_circuit_bytes),
+        }),
+        status: None,
+    };
+
+    let mut substream = Framed::new(io, quick_protobuf_codec::Codec::new(MAX_MESSAGE_SIZE));
+
+    let send_res = substream.send(msg).await;
+    if send_res.is_err() {
```

Comment:
Do we (still) need this split where we have to assign the result or can we just inline this variable?

```suggestion
    if substream.send(msg).await.is_err() {
```